### ISSUE TITLE
Multiple subsystems per reaction

### ISSUE THREAD

**Issue by FaribaRoshanzamir** [SysBioChalmers/Human-GEM#356]
### Description of the issue:
Two different subsystems (Heme synthesis and Porphyrin metabolism) are assigned to reaction "MAR01044". 
It seems that by merging two reactions in the commit https://github.com/SysBioChalmers/Human-GEM/commit/5f59d4262ba662887f5fbf78e1a817749b648ae4 , both subsystems were kept. 
I think it would be easier to decide which one should stay if we want to follow one subsystem per reaction in the model.  

#### Expected feature/value/output:
Having one subsystem for each reaction.

#### Current feature/value/output:
Reaction ID:  MAR01044
Reaction equation: Fe2+ [m] + protoporphyrin [m] ⇒ 2 H+ [m] + heme [m]
Gene rule:  FECH
Compartment(s):  Mitochondria
Subsystem(s):  Heme synthesis; Porphyrin metabolism

#### Reproducing these results:
```Matlab
 a = ihuman.subSystems;
 I=[];for i=1:length(a);c=size(a{i});if c(1)>1;rows=c(1);disp(rows);I=[I,i];end;end

```

**I hereby confirm that I have:**

- [X] Checked that a similar issue does not exist already

**Comment by mihai-sysbio** [comment]
I'm going to ping @haowang-bioinfo since he was mentioned in the original thread in #283. I suspect also @JonathanRob might have some thoughts on the matter too.

**Comment by JonathanRob** [comment]
I guess it's technically valid (as far as SBML is concerned) to have multiple subsystems per reaction, but since we have so far been using 1 to 1 relationships, I think it makes sense to keep it that way for now. 

I think it would be great to have a more exhaustive mapping of reactions to different subsystems, but that will take a fair amount of work and some consideration on how to implement.

If anyone is looking for a project to embark on - tackling this would be a much appreciated effort.

**Comment by mihai-sysbio** [comment]
Since this a breaking issue, I just labeled it as a bug. It might be nice to sort this out in time for the next release.

@FaribaRoshanzamir do you have a suggestion of which subsystem to keep and which to remove for the reaction `MAR01044`? Also, since only a small change is expected, it's a great way to experiment with creating pull requests. The [Contributing guidelines](https://github.com/SysBioChalmers/Human-GEM/blob/main/.github/CONTRIBUTING.md#contributing-to-the-modelannotation) have been recently improved. If you have any doubts just reach out. It's also a good opportunity to get some fresh eyes looking at the guide.


> great to have a more exhaustive mapping of reactions to different subsystems

@JonathanRob should there be another issue for this? Or a discussion? I'm willing to get involved as long as the process is scripted/automatic.

**Comment by haowang-bioinfo** [comment]
as @JonathanRob mentioned, having multiple subsystems is technically fine. @FaribaRoshanzamir which subsystem do you suggest, if only one were kept?

**Comment by haowang-bioinfo** [comment]
> @haowang-bioinfo I went through some of the databases. As I could see different pathways are assigned to reaction "MAR01044" with EC number 4.99.1.1.
It's assigned to "Porphyrin metabolism" in KEGG database, and "Heme metabolism" in BRENDA and MetaCyc databases.
"Heme biosynthesis " is only a pathway module of "Porphyrin metabolism" in KEGG and the reaction "MAR01044" is the last reaction for heme biosynthesis.
I don't have a strong opinion but since these two subsystems are connected closely, it would be fine if we just keep the "Porphyrin metabolism" subsystem.
Based on the model "https://metabolicatlas.org/explore/Human-GEM/gem-browser/subsystem/heme_synthesis" there are only four reactions assigned to Heme synthesis. So I thought we could use the same strategy as KEGG and just assign all of these reactions to "Porphyrin metabolism". What do you think @JonathanRob ?

@FaribaRoshanzamir thanks for the investigation, it seems a reasonable move to change from "Heme biosynthesis" to "Porphyrin metabolism" for the 4 reactions including `MAR01044`

**Comment by haowang-bioinfo** [comment]
fixed in #703
